### DISCUSSION TITLE
Video 78 - Evaluation our model's prediction on straight line.

### DISCUSSION THREAD

**Discussion by dsnal** [mrdbourke/pytorch-deep-learning#90] [Q&A] [answered]
In above video, I noticed that model_2.train( ) is not included epoch loop or above. Before that model, model.train is included in every model epoch loop. Why is that not included?

**Comment by mrdbourke** [comment, marked as answer]
Hey @dsnal, 

Good question!

`model.train()` is the default state of the model.

So it doesn't always have to be called but it should usually be called *after* calling `model.eval()`.

For example (the code is just an example and not a full training/testing loop):

```python
# Training loop
model.train() # this is the default
for x in training_data:
    model(x)

# Testing loop
model.eval()
for x in test_data:
    model(x)
```

So if you're training: call `model.train()`.

And if you're testing/making predictions: call `model.eval()`

**Reply by Sarthak-Singh04** [reply]
I had a same problem thanks for answering this question
